pkg/controllers/vdb/replicatedupgrade_reconciler.go

```diff
@@ -290,3 +257,44 @@ func (r *ReplicatedUpgradeReconciler) genNewSubclusterName(baseName string, scMa
 	}
 	return "", errors.New("failed to generate a unique subcluster name")
 }
+
+// duplicateSubcluster will return a new vapi.Subcluster that is based on
+// baseSc. This is used to mimic the primaries in replica group B.
+func (r *ReplicatedUpgradeReconciler) duplicateSubcluster(baseSc *vapi.Subcluster, newSCName, oldImage string) *vapi.Subcluster {
```

Comment:
If this function is used only for replica group B can you give it a more specific name like `duplicateSubclusterInReplicaGroupB` or `duplicateSubclusterForReplicaGroupB`